messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+                        )
+                        .setParameter("processingGroup", processingGroup)
+                        .setParameter("processingStartedLimit", getProcessingStartedLimit())
+                        .getResultList());
+    }
+
+    /**
+     * Finds the next dead letter after the provided {@code oldLetter} in the database. This is the message in the same
+     * {@code processingGroup} and {@code sequence}, but with the next index.
+     *
+     * @param oldLetter The base letter to search from.
+     * @return The next letter to process.
+     */
+    private DeadLetterEntry findNextDeadLetter(JpaDeadLetter<M> oldLetter) {
+        return transactionManager.fetchInTransaction(() -> {
+            try {
+                return entityManager()
+                        .createQuery(
+                                "SELECT dl FROM DeadLetterEntry dl "
+                                        + "where dl.processingGroup=:processingGroup "
+                                        + "and dl.sequence=:sequence "
+                                        + "and dl.index > :previousIndex "
+                                        + "order by dl.index asc ",
+                                DeadLetterEntry.class
+                        )
+                        .setParameter("processingGroup", processingGroup)
+                        .setParameter("sequence", oldLetter.getSequence())
+                        .setParameter("previousIndex", oldLetter.getIndex())
+                        .getSingleResult();
+            } catch (NoResultException exception) {
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Claims the dead provided {@link DeadLetter} in the database by setting the {@code processingStarted} property.
+     * Will check whether it was claimed successfully, or throw a {@link DeadLetterClaimException}.
+     */
+    private void claimDeadLetter(JpaDeadLetter<M> deadLetter) {
+        Instant processingStartedLimit = getProcessingStartedLimit();
+        transactionManager.executeInTransaction(() -> {
+            int updatedRows = entityManager().createQuery("update DeadLetterEntry dl set dl.processingStarted=:time "
+                                                                  + "where dl.deadLetterId=:deadletterId "
+                                                                  + "and (dl.processingStarted is null or dl.processingStarted < :processingStartedLimit)")
+                                             .setParameter("deadletterId", deadLetter.getId())
+                                             .setParameter("time", GenericDeadLetter.clock.instant())
+                                             .setParameter("processingStartedLimit", processingStartedLimit)
+                                             .executeUpdate();
+            if (updatedRows == 0) {
+                throw new DeadLetterClaimException(String.format(
```

Comment:
If this bubbles up from the `process` method, I would expect a test case in the `JpaSequencedDeadLetterQueueTest` class.